src/core/UserIconGroup/UserIconGroup.tsx

```diff
@@ -0,0 +1,121 @@
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import { useTheme } from '../utils/hooks/useTheme';
+import cx from 'classnames';
+import { UserIcon } from '../UserIcon';
+import { CommonProps } from '../utils/props';
+
+export type UserIconGroupProps = {
+  /**
+   * UserIconGroup max number of icons unstacked
+   * @default 5
+   */
+  maxIcons?: number;
+  /**
+   * UserIconGroup is group needs to be stacked
+   * @default 'true'
+   */
+  stacked?: boolean;
+  /**
+   * UserIconGroup type
+   * @default 'false'
+   */
+  animated?: boolean;
+  /**
+   * User Icons in the UserIconGroup.
+   */
+  children: React.ReactNode;
+} & CommonProps;
+
+/**
+ * Group User Icons together
+ * User Icons stacking is based on maxIcons count. If you provide 8 UserIcons and keep default 5 maxIcons count,
+ * this component will show 5 UserIcons and stacked UserIcon with "+3" in it.
+ *
+ * @example
+ * <UserIconGroup>
+ *  <UserIcon
+ *    abbreviation="TR"
+ *    backgroundColor="#6AB9EC"
+ *    size="medium"
+ *    title="Terry Rivers"
+ *    type="default"
+ *  />
+ *  <UserIcon
+ *   abbreviation="RM"
+ *    backgroundColor="#C8C2B4"
+ *    size="medium"
+ *    title="Robin Mercer"
+ *    type="default"
+ *  />
+ *  <UserIcon
+ *    abbreviation="TR"
+ *    backgroundColor="#6AB9EC"
+ *    size="medium"
+ *    title="Terry Rivers"
+ *    type="default"
+ *  />
+ * </UserIconGroup>
+ */
+export const UserIconGroup = (props: UserIconGroupProps) => {
+  const defaultLength = 5;
+  const maxLength = 9;
+  const {
+    children,
+    animated = false,
+    stacked = true,
+    maxIcons = defaultLength,
+  } = props;
+
+  const childrenArray = React.Children.toArray(children);
+  const childrenLength = React.Children.count(children);
+  let additionalChildren = '';
+  if (childrenLength > maxIcons) {
+    React.Children.forEach(children, (child: JSX.Element, index) => {
+      if (index >= maxIcons) {
+        additionalChildren += child.props.title + '\n';
+      }
+    });
+  }
+
+  useTheme();
+
+  return (
+    <div
+      className={cx('iui-user-icon-list', {
+        'iui-animated': animated,
+      })}
+    >
+      {childrenLength <= maxIcons + 1 || !stacked ? (
+        children
+      ) : (
+        <>
+          {childrenArray.map((child, index) =>
+            index < maxIcons
+              ? React.cloneElement(child as JSX.Element, {
+                  status: undefined,
+                  key: index,
+                })
+              : null,
+          )}
+
+          <UserIcon
+            size={(childrenArray[0] as JSX.Element).props.size}
+            title={additionalChildren}
+            abbreviation={
+              childrenLength <= maxLength
+                ? `+${childrenLength - maxIcons}`
+                : `${maxLength}+`
+            }
+            backgroundColor='#dce0e3'
+          />
```

Comment:
> User Icon takes only 2 symbols for abbreviation. Cant make it show 10+

This is by design.
https://github.com/iTwin/iTwinUI-react/blob/1b3488d3cae8bb22263947c780ba8e2e6ad5152b/src/core/UserIcon/UserIcon.tsx#L110-L112

Since we need to add a new class (https://github.com/iTwin/iTwinUI/pull/177) and support custom events anyway, I'd say don't use `UserIcon` and instead go with a custom div.

```suggestion
          <div
            {...excessIconProps} // this should be a new prop
            className={cx(
              'iui-user-icon',
              { [`iui-${size}`]: (childrenArray[0] as JSX.Element).props.size !== 'medium' },
              'iui-user-icon-count',
              excessIconProps.className
            )}
          >
            <abbr className='iui-initials'>
              {childrenLength <= maxLength ? `+${childrenLength - maxIcons}` : `${maxLength}+`}
            </abbr>
            <span className='iui-stroke' />
          </div>
```

Reply:
`excessIconProps` will add support for custom events if it's extending native div attributes, but I think we need one more prop to add support for tooltips/dropdowns 🤔 And then it starts getting complicated. Maybe we just need to allow passing a custom icon (which user can wrap in tooltip/dropdown) and show a storybook example.

---

AntD just forces a Popover, and has a `maxPopoverPlacement` prop. https://ant.design/components/avatar/#Avatar.Group-(4.5.0+)

Reply:
Another problem is that `UserIcon` does not forward ref, so popover won't show at all. We could add a wrapper but that breaks styling.